src/datamodel/builder/Uuid.cpp

```diff
@@ -52,50 +48,23 @@ UuidBuilder& UuidBuilder::withRandomSource(
 	return *this;
 }
 
-UuidBuilder& UuidBuilder::withIndependentState() {
-	if (!testing_) {
-		throw std::domain_error(
-		    "Cannot set independent state on non-test UuidBuilder");
-	}
-	shared_state_ = std::make_shared<UuidSharedState>();
-	return *this;
-}
-
 v1::UUID UuidBuilder::build() {
 	v1::UUID uuid;
 	auto now = time_source_ ? time_source_() : std::chrono::system_clock::now();
 	auto unix_ts_ms =
 	    std::chrono::time_point_cast<std::chrono::milliseconds>(now);
-
-	if (unix_ts_ms != shared_state_->last_unix_ts_ms) {
-		// Reset the counter if the timestamp tick has changed
-		shared_state_->counter = 0;
-		shared_state_->last_unix_ts_ms = unix_ts_ms;
-	}
+	std::mt19937_64 gen{std::random_device{}()};
+	std::uniform_int_distribution<uint16_t> dist_a(0, UUID_RANDOM_A_MASK);
+	std::uniform_int_distribution<uint64_t> dist_b(0, UUID_RANDOM_B_MASK);
```

Comment:
Minor: make these variables members of the `UuidBuilder` class. They can then be reused for each call to build and, if the `UuidBuilder` instance is held, the caller will get a little bit of performance benefit on each call to `build()`.